thor/src/protocols/splice.rs

```diff
@@ -100,21 +100,27 @@ impl State0 {
         previous_tx_f: FundingTransaction,
         x_self: OwnershipKeyPair,
         X_other: OwnershipPublicKey,
-        splice_in: Option<Amount>,
+        splice_self: crate::Splice,
```

Comment:
:upside_down_face: I find the use of these two types with the same name a bit confusing.